src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -195,8 +198,10 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
     return;
   }
 
-  // TODO: check forward_payload_header.
-  headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
+  LowerCaseString key(issuer_item.jwt_config().forward_payload_header());
```

Comment:
const

Reply:
Please keep original line 199.

then check forward_payload_header, if not empty and set the same as "sec-istio-auth-userinfo" set it.

Reply:
Done.